### ISSUE TITLE
Why need unnormalize img when apply openai's clip `embed_image`

### ISSUE THREAD

**Issue by CiaoHe** [lucidrains/DALLE2-pytorch#96]
Hi Phil,
I doubt whether need translate image from (-1,1) -> (0,1) here. Since the image parsed in is from the original dataset(PIL), and its range is (0,1), I think here should avoid using `unnormalize()`
https://github.com/lucidrains/DALLE2-pytorch/blob/2db0c9794c33e98df25b84f557a683a8900dfc61/dalle2_pytorch/dalle2_pytorch.py#L281

When I do an experiment on decoder training, the sampled image from `Decoder` turns out too bright (the whole pixel value shifts to a higher range)

Best,

**Comment by lucidrains** [comment]
@CiaoHe Hi again 👋 ! So actually, when training DDPMs, images have to be normalized from `-1 to 1` for the noising to work 

I don't have an example of image data loading in the repo yet, but it will look something like https://github.com/lucidrains/denoising-diffusion-pytorch/blob/master/denoising_diffusion_pytorch/denoising_diffusion_pytorch.py#L520

**Comment by lucidrains** [comment]
so the `embed_image` there is to actually act on the normalized images for ddpm training, but I think I should offer an option to turn off the unnormalization, in case it is used outside of the `Decoder` class

let me do that now actually

**Comment by lucidrains** [comment]
@CiaoHe actually i think you are right that that function shouldn't be in there. i've moved it into the decoder and diffusion prior, unnormalizing the image from [-1, 1] to [0, 1] just before sending it into the `embed_image` function

thank you for raising this issue!

**Comment by lucidrains** [comment]
actually come to think of it, i should just take care of the normalization and unnormalization all within the decoder class (and not in the dataloader) one less thing for people to worry about

**Comment by lucidrains** [comment]
come to think of it, should probably just take care of the normalization of image being trained and unnormalization of images being sampled within the decoder class automatically 🤔

**Comment by lucidrains** [comment]
ok all done in the latest! now nobody has to worry about this normalization / invers normalization business :)

**Comment by CiaoHe** [comment]
wow what a speed! I just took a snap  haha

**Comment by CiaoHe** [comment]
wow what a speed! I just took a snap  haha
